lms/security.py

```diff
@@ -31,31 +31,39 @@ class Permissions(Enum):
 class SecurityPolicy:
     """Top-level authentication policy that delegates to sub-policies."""
 
-    def __init__(self):
-        self._subpolicies = [LTISecurityPolicy(), LMSGoogleSecurityPolicy()]
+    def __init__(self, path_policy_mapping, fallback_policy):
+        """
+        Create a new SecurityPolicy.
+
+        :param path_policy_mapping: Map between route paths and policies
+        :param fallback_policy: Fallback policy in case no coincidences are found in path_policy_mapping
+        """
+        self.path_policy_mapping = path_policy_mapping
+        self.fallback_policy = fallback_policy
 
     def authenticated_userid(self, request):
-        return self._policy(request).authenticated_userid(request)
+        return self.get_route_policy(request).authenticated_userid(request)
 
     def identity(self, request):
-        return self._policy(request).identity(request)
+        return self.get_route_policy(request).identity(request)
 
     def permits(self, request, context, permission):
-        return self._policy(request).permits(request, context, permission)
+        return self.get_route_policy(request).permits(request, context, permission)
 
     def remember(self, request, userid, **kw):
-        return self._policy(request).remember(request, userid, **kw)
+        return self.get_route_policy(request).remember(request, userid, **kw)
 
     def forget(self, request):
-        return self._policy(request).forget(request)
+        return self.get_route_policy(request).forget(request)
 
     @lru_cache(maxsize=1)
-    def _policy(self, request):
-        for policy in self._subpolicies:
-            if policy.authenticated_userid(request):
-                return policy
+    def get_route_policy(self, request):
+        """Get the corresponding policy for this request's route."""
+        for path, policy in self.path_policy_mapping.items():
+            if request.path.startswith(path):
+                return policy()
 
-        return self._subpolicies[-1]
+        return self.fallback_policy()
```

Comment:
Fallback might not be required in the future, we might raise here.

Reply:
Yeah I was thinking that. If you added a `"/": LTISecurityPolicy` to the end of `path_policy_mapping` then I think you could get rid of `fallback_policy` now.

Reply:
:+1: A bit less explicit but adding a comment for that effect on the configuration side.